src/typechecker.erl

```diff
@@ -4042,6 +4098,10 @@ print_error(Error, Opts) ->
 handle_type_error({type_error, Expression, ActualType, ExpectedType}, Opts)
   when is_tuple(Expression) ->
     print_type_error(Expression, ActualType, ExpectedType, Opts);
+handle_type_error({nonexhaustive, Anno, Example}, Opts) ->
+    io:format("Nonexhaustive patterns starting on line ~p.~n"
+	      "Example values which is not covered: ~p~n"
+	     ,[format_location(Anno, verbose, Opts), Example]);
```

Comment:
Now, `--fmt-location verbose` works but running with `--location-fmt brief` doesn't show any line number at all.

The way to do this is slightly confusing, but if option `--fmt-location brief` is specified, `format_location(Anno, verbose, Opts)` returns nothing and `format_location(Anno, brief, Opts)` returns something. Vice versa if the option is `--fmt-location verbose`. The brief output is always the first in line since it's on the form ":line" or ":line:col" (to be shown directly after the filename, like the output of `grep -n`). Try with the CLI to be sure.

Reply:
Indeed it's currently quite inconvenient, but that was the only way I could implement to get the intended output :(